src/platform/forms-system/src/js/review/ObjectField.jsx

```diff
@@ -145,69 +145,72 @@ class ObjectField extends React.Component {
             .map(renderField);
         }
         return showReviewField(first, schema, uiSchema, formData, formContext)
-          ? // eslint-disable-next-line sonarjs/no-extra-arguments
-            renderField(first, index)
+          ? renderField(first)
           : null;
       },
     );
 
-    if (isRoot) {
-      let title = formContext.pageTitle;
-      if (!formContext.hideTitle && typeof title === 'function') {
-        title = title(formData, formContext);
-      }
-      const uiOptions = uiSchema['ui:options'] || {};
-      const ariaLabel = uiOptions.itemAriaLabel;
-      const itemName =
-        (typeof ariaLabel === 'function' && ariaLabel(formData || {})) ||
-        formData[uiOptions.itemName] ||
-        uiOptions.itemName;
-      const editLabel = (itemName && `Edit ${itemName}`) || `Edit ${title}`;
-
-      const Tag = divWrapper ? 'div' : 'dl';
-      const ObjectViewField = uiSchema?.['ui:objectViewField'];
-
-      const defaultEditButton = ({
-        label = editLabel,
-        onEdit = formContext.onEdit,
-        text = 'Edit',
-      } = {}) => (
-        <button
-          type="button"
-          className="edit-btn primary-outline"
-          aria-label={label}
-          onClick={onEdit}
-        >
-          {text}
-        </button>
-      );
+    let title = formContext?.pageTitle;
+    if (!formContext?.hideTitle && typeof title === 'function') {
+      // the `formData` is local to the object, not the page.
+      // A page would have access to properties that a child object wouldn't
+      title = isRoot && title(formData, formContext);
+    }
```

Comment:
Checking for `isRoot` before we call the `title` function is to prevent errors happening when a function that's expecting root-level data is passed object-level data.